src/components/AttachmentModal.js

```diff
@@ -178,11 +178,18 @@ function AttachmentModal(props) {
     }, []);
     /**
      * @param {Object} _file
+     * @param {Object} _data
      * @returns {Boolean}
      */
     const isValidFile = useCallback(
-        (_file) => {
+        (_file,_data) => {
```

Comment:
Why do we need both file and data? `_data.getAsFile` provides the file right?

Reply:
yeah, but this function is also used in the file picker addition to drag and drop, so there you won't get data files but normal file objects. Added this condition will resolve even if a user is uploading on different platforms it won't have an effect. 
we can remove `_data` props but we should add this check
```js
let _file = _data
if (typeof _data.getAsFile === 'function' ){
     _file = _data.getAsFile();
 }
```
so what's your suggestion?

Reply:
@abdulrahuman5196  

```js
 /**
     * @param {Object} _data
     * @returns {Object}
     */
    const isDirectoryCheckAndGetFile = useCallback((_data) => {
        if (typeof _data.webkitGetAsEntry === 'function' && _data.webkitGetAsEntry().isDirectory) {
            setIsAttachmentInvalid(true);
            setAttachmentInvalidReasonTitle(props.translate('attachmentPicker.attachmentError'));
            setAttachmentInvalidReason(props.translate('attachmentPicker.folderNotAllowedMessage'));
            return null;
        }
        if (typeof _data.getAsFile === 'function' ){
            return _data.getAsFile();
        }
        return _data

    },[props.translate])

const validateAndDisplayFileToUpload = ...(_data) => {
const _file = isDirectoryCheckAndGetFile(_data)

if(!_file){
return }
....

}...
```

what about this approach?

Reply:
IMO, IsDirectorycheck and getFile is different functionality. So IsDirectorycheck is fine and getFile can stay in current validateAndDisplayFileToUpload place

Reply:
done